File: haskell/problem-6/problem-6.hs
main = print answer

answer = (squareOfSum [1..100]) - (sumOfSquares [1..100])

squareOfSum :: [Integer] -> Integer
squareOfSum = square . sum

sumOfSquares :: [Integer] -> Integer
sumOfSquares = sum . map square

square :: Integer -> Integer
square n = n * n

File: go/problem-6/problem-6.go
package main

import (
    "fmt"
    "../packages/utilities/mathutils"
)

func main() {
    firstHundred := mathutils.NumbersBetween(1, 100)
    answer := squareOfSum(firstHundred) - sumOfSquares(firstHundred)

    fmt.Println(answer)
}

func squareOfSum(nums []int) int {
    var sum int

    for _, num := range nums {
        sum += num
    }

    return (sum * sum)
}

func sumOfSquares(nums []int) int {
    var sum int

    for _, num := range nums {
        square := num * num
        sum += square
    }

    return sum
}
